third/CMakeLists.txt

```diff
@@ -19,7 +23,105 @@ set(BGFX_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
 set(GLM_TEST_ENABLE OFF CACHE BOOL "" FORCE)
 add_subdirectory(bgfx)
 
+# ImGUI
+
+project(imgui)
+file(GLOB IMGUI_SOURCES
+  imgui/imgui.cpp
+  imgui/imgui_draw.cpp
+  imgui/imgui_tables.cpp
+  imgui/imgui_widgets.cpp
+  imgui/imgui_demo.cpp
+
+  imgui/backends/imgui_impl_glfw.cpp
+  imgui/backends/imgui_impl_opengl3.cpp
+)
+add_library(imgui ${IMGUI_SOURCES})
+target_link_libraries(imgui PRIVATE glfw)
+target_include_directories(imgui PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/imgui)
+target_include_directories(imgui PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/imgui/backends)
+
+# ImGuizmo
+
+project(imguizmo)
+file(GLOB IMGUIZMO_SOURCES
+  imguizmo/GraphEditor.cpp
+  imguizmo/ImCurveEdit.cpp
+  imguizmo/ImGradient.cpp
+  imguizmo/ImGuizmo.cpp
+  imguizmo/ImSequencer.cpp
+)
+add_library(imguizmo ${IMGUIZMO_SOURCES})
+target_link_libraries(imguizmo PRIVATE imgui)
+target_include_directories(imguizmo PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/imguizmo)
+
+# ImTerm
+
+project(imterm)
+add_library(imterm INTERFACE)
+
+target_include_directories(imterm INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/ImTerm/include)
+
+# JSON
+
+add_subdirectory(json)
+
+# PhysX
+
+if(DEFINED ENV{PHYSX_PATH})
+    set(PHYSX_ROOT_DIR ${ENV{PHYSX_PATH}})
+    message(STATUS "Using PhysX path: '${PHYSX_ROOT_DIR}'")
+else()
+    set(PHYSX_ROOT_DIR "${CMAKE_SOURCE_DIR}/third/physx/physx")
+    message(STATUS "Using default PhysX path: '${PHYSX_ROOT_DIR}'")
+endif()
+set(ENV{PHYSX_PATH} ${PHYSX_ROOT_DIR})
+
+set(PHYSX_INCLUDE_DIRS $ENV{PHYSX_PATH}/include/ $ENV{PHYSX_PATH}/../pxshared/include/ PARENT_SCOPE)
+set(PHYSX_LIBRARIES
+    PhysXExtensions
+    PhysX
+    PhysXPvdSDK
+    PhysXVehicle
+    PhysXCharacterKinematic
+    PhysXCooking
+    PhysXCommon
+    PhysXFoundation
+PARENT_SCOPE)
+
+# has to match the TARGET_BUILD_PLATFORM in $ENV{PHYSX_PATH}/physix/buildtools/cmake_generate_projects.py
+if(WIN32)
```

Comment:
Do we also need to check for x64?

Reply:
CMake naming convention is weird. WIN32 refers to both architectures, it just means it wants windows.

https://cmake.org/cmake/help/v3.0/variable/WIN32.html